obspy/io/nied/fnetmt.py

```diff
@@ -0,0 +1,254 @@
+    """
+    Test whether file is an F-net moment tensor catalog file by reading the
+    header and the first data line. Reads at most 40 lines.
+
+    :param buf: File to read.
+    :type buf: Open file or open file like object.
+    """
+    cnt = 0
+    try:
+        while True:
+            line = buf.readline()
+            if not line:
+                return False
+            line = line.decode()
+            # read at most 40 lines
+            if cnt > 40:
+                return False
+            if line.find('Total Number') != -1:
+                match = re.search(r'Total Number:\s+(\d+)\s+', line)
+                if match:
+                    nevents = int(match.group(1))
+            if line.startswith('Origin Time'):
+                if nevents > 0:
+                    data = buf.readline()
+                    a = data.split()
+                    if len(a) != 21:
+                        return False
+                    return True
+            cnt += 1
+    except:
+        return False
+    else:
+        return True
+
+
+def _read_fnetmt_catalog(filename_or_buf, **kwargs):
+    """
+    Reads an F-net moment tensor catalog file to a
+    :class:`~obspy.core.event.Catalog` object.
+
+    :param filename_or_buf: File to read.
+    :type filename_or_buf: str or file-like object.
+    """
+    return _buffer_proxy(filename_or_buf, __read_fnetmt_catalog, **kwargs)
+
+
+def __read_fnetmt_catalog(buf, **kwargs):
+    """
+    Reads an F-net moment tensor catalog file to a
+    :class:`~obspy.core.event.Catalog` object.
+
+    :param buf: File to read.
+    :type buf: Open file or open file like object.
+    """
+    events = []
+    cur_pos = buf.tell()
+
+    # This also works with BytesIO and what not.
+    buf.seek(0, 2)
+    size = buf.tell()
+    buf.seek(cur_pos, 0)
+
+    # First read the headerlines containing the data request parameters
+    headerlines = []
+    while buf.tell() < size:
+        line = buf.readline().decode()
+        if line.find('Total Number') != -1:
+            match = re.search(r'Total Number:\s+(\d+)\s+', line)
+            if match:
+                nevents = int(match.group(1))
+        if line.startswith('Origin Time'):
+            break
+        headerlines.append(line)
+    cur_pos = buf.tell()
+
+    # Now read the catalog
+    while buf.tell() < size:
+        line = buf.readline().strip()
+        # If there is something, jump back to the beginning of the line and
+        # read the next event.
+        if line:
+            events.append(__read_single_fnetmt_entry(line.decode()))
+
+    # Consistency check
+    if len(events) != nevents:
+        raise FNETMTException('Parsing failed! Expected %d events but read %d.'
+                              % (nevents, len(events)))
+
+    return Catalog(resource_id=_get_resource_id("catalog", str(uuid.uuid4())),
+                   events=events, description=headerlines[:-1])
+
+
+def __read_single_fnetmt_entry(line, **kwargs):
```

Comment:
I still don't know what the point of the `kwargs` is here; only one place calls this function and it never passes in any keyword arguments.